pep-0484.txt

```diff
@@ -448,6 +448,63 @@ is not generic but implicitly inherits from ``Iterable[Any]``::
 Generic metaclasses are not supported.
 
 
+Scoping rules for type variables
+--------------------------------
+
+Type variables follow normal name resolution rules.
+However, there are some special cases in the static typechecking context:
+
+* A type variable could be inferred to be equal to different types in a block,
```

Comment:
I tried to reformulate this by splitting it into three separate rules. Please check if it is more clear now.